src/components/SecondaryPanes/Scopes.js

```diff
@@ -96,12 +104,18 @@ class Scopes extends PureComponent<Props, State> {
 export default connect(
   state => {
     const selectedFrame = getSelectedFrame(state);
-    const frameScopes = selectedFrame
-      ? getFrameScope(state, selectedFrame.id)
-      : null;
+    const selectedSource = getSelectedSource(state);
+
+    const { scope: frameScopes, pending } = getFrameScope(
+      state,
+      selectedSource && selectedSource.get("id"),
+      selectedFrame.id
+    ) || { pending: false };
```

Comment:
should this `{ pending: false };` be in the selector?

so if we dont have scopes we just return `{ pending: false, scope: null };`

Reply:
I don't love the idea of returning something when it wasn't actually in the state. It also depends on if you want to differentiate between "loaded but not found", and "not loaded at all". You're right that we don't care here, but I don't really love making that assumption for all possible uses of `getFrameScope`.

Reply:
yeah - that is true.